net/wireguard/src/opnsense/mvc/app/models/OPNsense/Wireguard/Client.xml

```diff
@@ -13,12 +13,14 @@
                     <default></default>
                     <Required>Y</Required>
                 </name>
-                <pubkey type="TextField">
+                <pubkey type="Base64Field">
                     <Required>N</Required>
+                    <mask>/^[a-zA-Z0-9+=\/]{44}$/</mask>
```

Comment:
I don't think the mask is needed any longer as the type validates it, which leaves you with the length, but it seems like micromanagement to check it and break later when new key length is added

Reply:
Agree but I think this check is currently useful until the wall wireguard server stop hang if one public key of one peer doesn't match the required format. I just tried to provide a random base64 string with the wrong size setting up one test peer and it prevented the wall wireguard process to restart, not just the associated tunnel "server". It is a real issue which must be fix somewhere else. Keeping this checks here for now may help in the mean time 

What do you think?

Reply:
the base64 field was added to reduce complexity and typographic errors in manual regex masks. having the field and keeping the mask amounts do doing the base64 field work for very little benefit and keeps maintenance of these configuration files higher than they have to be.

also think of copy+paste development so bad style like this will get carried over if we don't keep on top of it. :)

Reply:
Yeah right, I also think this is enough check performed client-side, public key should be, somehow, checked server side and an issue may be opened. I mentioned in an other thread this PR #2008 which may help